File: Problems/Parking Lot System/Ticket.cpp
#include "./ParkingSpots.cpp"
#include "./Vechicle.cpp"

class Ticket {
    int ticketId;
    Vehicle *vehicle;
    ParkingSpot *spot;
    time_t entryTime;
    public:
    Ticket(int id,Vehicle *v,ParkingSpot *spot): ticketId(id),vehicle(v),spot(spot) {}

    int getTicketId() const { return ticketId; }

    Vehicle* getVehicle() const { return vehicle; }

    ParkingSpot* getSpot() const { return spot; }
    
    time_t getEntryTime() const { return entryTime; }
};

File: Problems/Parking Lot System/Ticket.java
package Problems.Java;

public class Ticket {
    Spot spot;
    Vehicle vehicle;
    long entryTime;
    long exitTime;
    double totalFare;
    public Ticket(Spot spot, Vehicle vehicle, long entryTime) {
        this.spot = spot;
        this.vehicle = vehicle;
        this.entryTime = entryTime;
    }

    Spot getSpot() {
        return spot;
    }

    Vehicle getVehicle() {
        return vehicle;
    }

}
